cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -592,6 +592,39 @@ VersionedItem LocalVersionedEngine::write_versioned_metadata_internal(
     }
 }
 
+std::vector<VersionedItem> LocalVersionedEngine::batch_write_versioned_metadata_internal(
+    std::vector<StreamId> stream_ids,
+    bool prune_previous_versions,
+    std::vector<arcticdb::proto::descriptors::UserDefinedMetadata>&& user_meta_proto_vec) {
+    auto stream_update_info_vector_fut = batch_get_latest_undeleted_version_and_next_version_id_async(store(),
+                                                                                                      version_map(),
+                                                                                                      stream_ids);
+    std::vector<folly::Future<VersionedItem>> output_vect_fut;
+    for(const auto& stream_id : folly::enumerate(stream_ids)) {
```

Comment:
I discovered yesterday that this syntax is also valid, and I think cleaner in this case:
`for (const auto&& [idx, stream_id] : folly::enumerate(stream_ids))`

Reply:
sounds plausible

Reply:
amended